Assets/Scripts/Spawner/SanitySpawnController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Entity.Enemy;
+using Entity.Player;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.AI;
+using Utils;
+using System.Linq;
+using System.Collections.Generic;
+using Boo.Lang;
+
+namespace Spawner
+{
+    public class SanitySpawnController : NPCSpawner
+    {
+        public delegate void PlayerStateChanged(int newValue);
+
+        [SerializeField]
+        private float spawnDimensionScale;
+
+        // at sanity = 45
+        [SerializeField]
+        private float despawnDimensionScale;
+
+        [SerializeField]
+        private AnimationCurve probabilityScaleCurve;
+
+        [Tooltip("Spawn probability is scaled with a factor 1-x for every active/existing sanity NPC.")]
+        [SerializeField]
+        private float activeSpawnPenalty;
+
+
+        private Dictionary<Renderer, GameObject> _spawned;
```

Comment:
```suggestion
        private Dictionary<Renderer, GameObject> _spawned = new Dictionary<Renderer, GameObject>();
```
No need to do it in Awake. Also, if you never overwrite it with a new dictionary, you can add the keyword `readonly` as well.